`torchWithRocm`still fails to build with this and the PR linked above:

```
FAILED: caffe2/CMakeFiles/torch_hip.dir/__/aten/src/ATen/hip/detail/HIPHooks.cpp.o 
/nix/store/kvlhk0gpm2iz1asbw1xjac2ch0r8kyw9-gcc-wrapper-13.2.0/bin/g++ -DAT_PER_OPERATOR_HEADERS -DHAVE_MALLOC_USABLE_SIZE=1 -DHAVE_MMAP=1 -DHAVE_SHM_OPEN=1 -DHAVE_SHM_UNLINK=1 -DMAGMA_V2 ->
cc1plus: warning: command-line option ‘-Wno-duplicate-decl-specifier’ is valid for C/ObjC but not for C++
In file included from /build/source/aten/src/ATen/core/IListRef.h:631,
                 from /build/source/aten/src/ATen/DeviceGuard.h:3,
                 from /build/source/aten/src/ATen/hip/detail/HIPHooks.cpp:6:
/build/source/aten/src/ATen/core/IListRef_inl.h: In static member function ‘static c10::detail::IListRefConstRef<at::OptionalTensorRef> c10::detail::IListRefTagImpl<c10::IListRefTag::Boxed,>
/build/source/aten/src/ATen/core/IListRef_inl.h:171:17: warning: possibly dangling reference to a temporary [-Wdangling-reference]
  171 |     const auto& ivalue = (*it).get();
      |                 ^~~~~~
/build/source/aten/src/ATen/core/IListRef_inl.h:171:35: note: the temporary was destroyed at the end of the full expression ‘(& it)->c10::impl::ListIterator<std::optional<at::Tensor>, __gnu>
  171 |     const auto& ivalue = (*it).get();
      |                          ~~~~~~~~~^~
/build/source/aten/src/ATen/hip/detail/HIPHooks.cpp: In member function ‘virtual bool at::cuda::detail::CUDAHooks::isPinnedPtr(const void*) const’:
/build/source/aten/src/ATen/hip/detail/HIPHooks.cpp:158:15: error: ‘hipPointerAttribute_t’ {aka ‘struct hipPointerAttribute_t’} has no member named ‘memoryType’
  158 |   return attr.memoryType == hipMemoryTypeHost;
      |               ^~~~~~~~~~
```

Maybe the correct fix for now is to switch to `rocmPackages_5`.

--- a/pkgs/development/python-modules/torch/default.nix
+++ b/pkgs/development/python-modules/torch/default.nix
@@ -52,13 +52,15 @@
 
   # ROCm dependencies
   rocmSupport ? config.rocmSupport,
-  rocmPackages,
+  rocmPackages_5,
   gpuTargets ? [ ]
 }:
 
 let
   inherit (lib) attrsets lists strings trivial;
   inherit (cudaPackages) cudaFlags cudnn nccl;
+
+  rocmPackages = rocmPackages_5;
 
   setBool = v: if v then "1" else "0";
 

